app/frontend/src/store/modules/auth.js

```diff
@@ -34,23 +41,46 @@ export default {
       }
       return false; // There are roles to check, but nothing in token to check against
     },
-    identityProvider: () => Vue.prototype.$keycloak.tokenParsed.identity_provider,
+    identityProvider: () =>
+      Vue.prototype.$keycloak.tokenParsed.identity_provider,
     isAdmin: (_state, getters) => getters.hasResourceRoles('chefs', ['admin']),
     isUser: (_state, getters) => getters.hasResourceRoles('chefs', ['user']),
     keycloakReady: () => Vue.prototype.$keycloak.ready,
     keycloakSubject: () => Vue.prototype.$keycloak.subject,
     moduleLoaded: () => !!Vue.prototype.$keycloak,
     realmAccess: () => Vue.prototype.$keycloak.tokenParsed.realm_access,
-    redirectUri: state => state.redirectUri,
+    redirectUri: (state) => state.redirectUri,
     resourceAccess: () => Vue.prototype.$keycloak.tokenParsed.resource_access,
     token: () => Vue.prototype.$keycloak.token,
     tokenParsed: () => Vue.prototype.$keycloak.tokenParsed,
-    userName: () => Vue.prototype.$keycloak.userName
+    userName: () => Vue.prototype.$keycloak.userName,
+
+    user: (_state, getters) => {
+      const username = getters.authenticated
+        ? getters.tokenParsed.identity_provider_identity
+          ? getters.tokenParsed.identity_provider_identity
+          : getters.tokenParsed.preferred_username
+        : '';
+
+      const user = {
+        username,
+        firstName: getters.authenticated ? getters.tokenParsed.given_name : '',
+        lastName: getters.authenticated ? getters.tokenParsed.family_name : '',
+        fullName: getters.authenticated ? getters.tokenParsed.name : '',
+        email: getters.authenticated ? getters.tokenParsed.email : '',
+        idp: getters.authenticated
+          ? getters.tokenParsed.identity_provider
+          : 'public',
+        public: !!getters.authenticated,
+      };
```

Comment:
Not quite too sure what this getter function is needed for - we're constructing an object that a user can easily access directly by referencing token (tokenParsed) on formio side and going for the raw data directly if available.

If we do need this, should we consider having the default be assigned as undefined instead of an empty string? It's generally better to have an attribute be undefined if they're falsy.

Reply:
formio was also reading a user context from localstorage, so I created this to block any potential reads to localstorage without a `user` variable existing. The `user` variable could also be easier to reason about for the end user and isn't so reliant on any token structure itself.

Reply:
https://github.com/formio/formio.js/blob/73bb21a6d158db2d6eb33250ceff2997390ac398/src/Element.js#L549

Reply:
As for undefined, more parsing would then be needed on both form viewer and designer to set to `''` so that the text inputs get a good default value.